File: common/validation/validator.go
package validation

import (
	"github.com/go-playground/validator/v10"
)

func Validate(data interface{}) map[string]interface{} {
	validate := validator.New()
	errors := map[string]interface{}{}

	if err := validate.Struct(data); err != nil {
		if _, ok := err.(*validator.InvalidValidationError); ok {
			errors["base"] = err
			return errors
		}

		for _, err := range err.(validator.ValidationErrors) {
			errors[err.Field()] = err.Error()
		}
	}
	return errors
}

File: libs/Validator.js
const ValidatorJSClass = require('validatorjs');
const Message = require('../constants/Messages');

/**
 * An adapter of validatorjs library.
 */
class Validator {
  /**
   * Ensure the data matches all rules.
   *
   * @param {object} data
   * @param {object} rules
   * @param {object} messages
   * @return {void}
   *
   * @throws {Error}
   */
  validate(data, rules, messages = {}) {
    const v = new ValidatorJSClass(data, rules, messages);

    if (v.fails()) {
      const error = new Error(Message.UNPROCESSABLE_ENTITY);
      error.errors = {};
      for (const field in rules) {
        if (v.errors.has(field)) {
          error.errors[field] = v.errors.first(field);
        }
      }
      throw error;
    }
  }
}

module.exports = new Validator();
